src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -131,15 +133,31 @@ private void initializeHandlers() {
     }
 
     private void subscribeToConfigChanges() {
+        // This first call is just to initialize an empty configuration
         onConfigurationChanged();
         config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe(this::configChangeHandler);
     }
 
     private void onConfigurationChanged() {
-        try {
-            cdaConfiguration = CDAConfiguration.from(cdaConfiguration, getConfig());
-        } catch (URISyntaxException e) {
-            serviceErrored(e);
+        Result<CDAConfiguration> result = CDAConfiguration.from(cdaConfiguration, getConfig());
```

Comment:
This will return a result with a configuration regardless of wether it has errored or not, we want to track the configuration even if it is bad so that we can check if it has changed from a bad one to a good one or from a bad one to another bad one.

Reply:
It doesn't seem like most of these changes are necessary. There's really no need to track bad configuration, is there? If configuration is null and state is broken, request reinstall.